### PR TITLE
non-restricted bot role

### PR DESCRIPTION
**Pull request by joshblum**
non restricted bots have access to the PTK and are treated close to the existing `READER` role

cc @keybase/hotpotatosquad

### CODE REVIEW

File: go/protocol/keybase1/extras.go [hunk 1]
```diff
@@ -2324,10 +2330,18 @@ func (r TeamRole) IsReaderOrAbove() bool {
...
 
+func (r TeamRole) IsBotOrAbove() bool {
+	return r.IsOrAbove(TeamRole_BOT)
+}
+
 func (r TeamRole) IsRestrictedBotOrAbove() bool {
 	return r.IsOrAbove(TeamRole_RESTRICTEDBOT)
 }
 
+func (r TeamRole) IsBot() bool {
```
Comment (mlsteele) [inline go/protocol/keybase1/extras.go]: This name could be easily misunderstood as `r == BOT || r == RESTRICTEDBOT`. Might be less trouble to use `==` at the callsite.

(Or names `IsBotOfAnySort` `IsExactlyBot`)

File: go/protocol/keybase1/extras.go [hunk 2]
```diff
@@ -2337,9 +2351,23 @@ func (r TeamRole) IsOrAbove(min TeamRole) bool {
...
+		default:
+			return false
+		}
 	default:
-		return int(r) >= int(min) || min == TeamRole_RESTRICTEDBOT
+		return (int(r) >= int(min) ||
+			min == TeamRole_BOT ||
+			min == TeamRole_RESTRICTEDBOT)
 	}
 }
```
Comment (mlsteele) [inline go/protocol/keybase1/extras.go]: This method is getting tricky. How about using very-private int values like this? https://goplay.space/#-7jmf09gPbX

File: go/teams/rpc_exim.go [hunk 3]
```diff
@@ -45,6 +45,7 @@ func (t *Team) ExportToTeamPlusApplicationKeys(ctx context.Context, idTime keyba
 	writers = append(writers, members.Admins...)
 	writers = append(writers, members.Owners...)
 	onlyReaders = append(onlyReaders, members.Readers...)
+	onlyReaders = append(onlyReaders, members.Bots...)
```
Comment (mlsteele) [inline go/teams/rpc_exim.go]: onlyReaders -> onlyBots
which would then be unused

File: go/protocol/keybase1/extras.go (hunk 1 above)
Comment (joshblum) [reply go/protocol/keybase1/extras.go]: Oh the only place we use it is where we want `BotLike` so i'll convert to that

File: go/protocol/keybase1/extras.go (hunk 2 above)
Comment (joshblum) [reply go/protocol/keybase1/extras.go]: yeah much better

File: go/teams/rpc_exim.go (hunk 3 above)
Comment (joshblum) [reply go/teams/rpc_exim.go]: this would exclude `Bots` from being able to read kbfs (only one that uses this afaik) which they should be able to

Comment (mlsteele) [reply go/teams/rpc_exim.go]: oh ok

File: go/libkb/chain_link_v2.go [hunk 4]
```diff
@@ -157,7 +157,7 @@ func (t SigchainV2Type) RequiresAtLeastRole() keybase1.TeamRole {
 }
 
 func (t SigchainV2Type) TeamAllowStubWithAdminFlag(isAdmin bool) bool {
-	role := keybase1.TeamRole_READER
+	role := keybase1.TeamRole_BOT
```
Comment (mlsteele) [inline go/libkb/chain_link_v2.go]: There's something pre-existingly weird about these methods. I think `TeamAllowStub` is unused. If you'd like you can merge them and I think `role` goes away.

File: go/protocol/keybase1/extras.go [hunk 5]
```diff
@@ -2324,25 +2330,48 @@ func (r TeamRole) IsReaderOrAbove() bool {
...
+	case TeamRole_BOT:
+		return 2
+	case TeamRole_READER,
+		TeamRole_WRITER,
+		TeamRole_ADMIN,
+		TeamRole_OWNER:
+		return int(r) + 2
 	default:
-		return int(r) >= int(min) || min == TeamRole_RESTRICTEDBOT
+		panic("unhandled teamRole")
```
Comment (mlsteele) [inline go/protocol/keybase1/extras.go]: woah there how about 0?

Comment (joshblum) [reply go/protocol/keybase1/extras.go]: oh i thought it would be better to crash since this is a sensitive check

Comment (mlsteele) [reply go/protocol/keybase1/extras.go]: Previously it didn't panic and just returned the wrong result. An error return path would be one way to go. I think the panic would do more harm than good imagining some ancillary system that using the comparison on untrusted input. Like some chat doohickey where someone sends a system message "Josh joined the team as a role 10".

Comment (joshblum) [reply go/protocol/keybase1/extras.go]: Heh yeah chat already uses this in a few places, good call. Will switch back to 0

**Pull request merged**